test/dymaptic.GeoBlazor.Core.Test.Blazor.Shared/Components/FeatureLayerTests.razor

```diff
@@ -470,4 +470,361 @@
+
+        await layer!.SetMaxScale(0);
+        Assert.AreEqual(0, await layer.GetMaxScale());
+        await AssertJavaScript("featureLayerAsserts.assertFeatureLayerNumberPropEquals", args: [layer!.Id, "maxScale", 0]);
+
+        await layer!.SetRefreshInterval(0.1);
+        Assert.AreEqual(0.1, await layer.GetRefreshInterval());
+        await AssertJavaScript("featureLayerAsserts.assertFeatureLayerNumberPropEquals", args: [layer!.Id, "refreshInterval", 0.1]);
+
+        await layer!.SetDefinitionExpression("1=1");
+        Assert.AreEqual("1=1", await layer.GetDefinitionExpression());
+        await AssertJavaScript("featureLayerAsserts.assertFeatureLayerStringPropEquals", args: [layer!.Id, "definitionExpression", "1=1"]);
+
+        await layer!.SetDateFieldsTimeZone("UTC");
+        Assert.AreEqual("UTC", await layer.GetDateFieldsTimeZone());
+        await AssertJavaScript("featureLayerAsserts.assertFeatureLayerStringPropEquals", args: [layer!.Id, "dateFieldsTimeZone", "UTC"]);
+
+        await layer!.SetDisplayField("OBJECTID");
+        Assert.AreEqual("OBJECTID", await layer.GetDisplayField());
+        await AssertJavaScript("featureLayerAsserts.assertFeatureLayerStringPropEquals", args: [layer!.Id, "displayField", "OBJECTID"]);
+    }
+
+    [TestMethod]
+    public async Task FeatureLayer_Collections_AddRemove_Roundtrip(Action renderHandler)
+    {
+        var initialOutFields = new[] { "Field1", "OBJECTID" };
+        FeatureLayer? layer = null;
+
+        AddMapRenderFragment(
+            @<MapView class="map-view" OnViewRendered="renderHandler">
+                <Map>
+                    <Basemap>
+                        <BasemapStyle Name="BasemapStyleName.ArcgisTopographicBase" />
+                    </Basemap>
+
+                    <FeatureLayer @ref="layer"
+                                  Source="@([])"
+                                  OutFields="@(initialOutFields)"
+                                  SpatialReference="SpatialReference.Wgs84"
+                                  GeometryType="FeatureGeometryType.Point"
+                                  ObjectIdField="OBJECTID">
+                        <Field Name="Field1"   Type="FieldType.Long"    />
+                        <Field Name="Field2"   Type="FieldType.Date"    />
+                        <Field Name="Field3"   Type="FieldType.String"  />
+                        <Field Name="OBJECTID" Type="FieldType.Integer" />
+                    </FeatureLayer>
+                </Map>
+            </MapView>);
+
+        await WaitForMapToRender();
+        await AssertJavaScript("assertLayerExists", args: "feature");
+
+        await layer!.SetOutFields(new[] { "Field1", "OBJECTID", "Field2" });
+
+        var outFields = await layer.GetOutFields();
+        CollectionAssert.AreEquivalent(new[] { "Field1", "OBJECTID", "Field2" }, outFields?.ToArray());
```

Comment:
There might be a few like these collection asserts that won't work with `assertObjectHasPropertyWithValue` and need to keep the custom functions in `featureLayerAsserts.js`.